samples/extensions/layer_settings/layer_settings.cpp

```diff
@@ -0,0 +1,315 @@
+			{
+				LOGW("Failed to create local debug messenger (res=%d)", int(res));
+			}
+		}
+	}
+
+	return inst;
+}
+
+LayerSettingsSample::LayerSettingsSample()
+{
+	title = "Layer settings (VK_EXT_layer_settings)";
+
+	// Configure the Khronos validation layer using layer settings. These settings are
+	// consumed by the validation layer at instance creation time.
+	//
+	// Note: The settings only take effect if the layer is enabled (e.g. by building
+	// with validation layers on, or enabling them via the application's options).
+	//
+	// 1) Enable Best Practices (generic + vendor-specific)
+	{
+		static const char *enables[] = {
+		    "VK_VALIDATION_FEATURE_ENABLE_BEST_PRACTICES_EXT",
+		};
+
+		VkLayerSettingEXT layer_setting{};
+		layer_setting.pLayerName   = "VK_LAYER_KHRONOS_validation";
+		layer_setting.pSettingName = "enables";
+		layer_setting.type         = VK_LAYER_SETTING_TYPE_STRING_EXT;
+		layer_setting.valueCount   = static_cast<uint32_t>(std::size(enables));
+		layer_setting.pValues      = enables;
+		add_layer_setting(layer_setting);
+	}
+
+	// 2) Optionally enable debug printf so shaders using debugPrintfEXT will print via VVL
+	{
+		static const char *enables[] = {
+		    "VK_VALIDATION_FEATURE_ENABLE_DEBUG_PRINTF_EXT",
+		};
+
+		VkLayerSettingEXT layer_setting{};
+		layer_setting.pLayerName   = "VK_LAYER_KHRONOS_validation";
+		layer_setting.pSettingName = "enables";
+		layer_setting.type         = VK_LAYER_SETTING_TYPE_STRING_EXT;
+		layer_setting.valueCount   = static_cast<uint32_t>(std::size(enables));
+		layer_setting.pValues      = enables;
+		add_layer_setting(layer_setting);
+	}
+
+	// 3) Demonstrate disabling a known verbose message category (example)
+	//    This shows how to filter messages with the "disables" setting.
+	//    Replace with a concrete message enable/disable as needed in your environment.
+	{
+		static const char *disables[] = {
+		    // Example: Treat performance warnings as disabled (users can customize)
+		    "VK_VALIDATION_FEATURE_DISABLE_ALL_EXT"        // users can comment this out; kept as illustration
+		};
+
+		VkLayerSettingEXT layer_setting{};
+		layer_setting.pLayerName   = "VK_LAYER_KHRONOS_validation";
+		layer_setting.pSettingName = "disables";
+		layer_setting.type         = VK_LAYER_SETTING_TYPE_STRING_EXT;
+		layer_setting.valueCount   = static_cast<uint32_t>(std::size(disables));
+		layer_setting.pValues      = disables;
+		// Do not add this by default as it disables all validation. Leave as commented example.
+		// add_layer_setting(layer_setting);
+	}
+}
+
+bool LayerSettingsSample::prepare(const vkb::ApplicationOptions &options)
+{
+	if (!ApiVulkanSample::prepare(options))
+	{
+		return false;
+	}
+
+	// Install a local DebugUtils messenger to collect validation messages into the UI.
+	// If it was already created during create_instance(), skip to avoid duplication.
+	if (debug_messenger_ == VK_NULL_HANDLE && get_instance().is_enabled(VK_EXT_DEBUG_UTILS_EXTENSION_NAME))
```

Comment:
I don't get this.
Under what circumstances would the `debug_messenger_` not be created during `create_instance`, which is called before this function?